Refactored metrics-related functions from framework/metrics_util.go

alejandrox1 opened the pull request:
This a refactoring of framework/metrics_utils.go into framework/metrics.

Signed-off-by: alejandrox1 <alarcj137@gmail.com>

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
6. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This is intended to clean up the e2e framework by refactoring framework/metrics_utils.go into framework/metrics.

**Which issue(s) this PR fixes**:
Part of #76206

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
NONE
```

/area e2e-test-framework
/area test
/sig testing
/priority backlog

alejandrox1 committed:
Refactored metrics-related functions from framework/metrics_util.go

This a refactoring of framework/metrics_utils.go into framework/metrics.

Signed-off-by: alejandrox1 <alarcj137@gmail.com>

alejandrox1 commented:
/assign @andrewsykim @timothysc @neolit123 
ptal

alejandrox1 commented:
```
Errors from golint:
test/e2e/framework/metrics/e2e_metrics.go:37:6: type name will be used as metrics.MetricsForE2E by other packages, and that stutters; consider calling this ForE2E
```

What do yall think about the naming for this one?

neolit123 commented on test/e2e/framework/kubelet_stats.go:
```diff
@@ -47,6 +47,7 @@ import (
 
 // KubeletLatencyMetric stores metrics scraped from the kubelet server's /metric endpoint.
 // TODO: Get some more structure around the metrics and this type
+// TODO(alejandrox1): this is already present in test/e2e/framework/metrics.
```
i'm assuming these will come in a separate PR?

i don't know about the others but i'm in favor of having multiple commits per PR.
this reduces my notifications spam and is still easy to review, because one can review single commit at a time.

timothysc replied:
Yeah, but this PR is already big enough, when you add the TODOs please open an issue.

alejandrox1 replied:
created https://github.com/kubernetes/kubernetes/issues/79907

neolit123 reviewed (commented):
/approve
thanks @alejandrox1. added one comment about the possibility of having multiple commits.

timothysc commented on test/e2e/framework/metrics/api.go:
```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2015 The Kubernetes Authors.
```
b/c it's a new file update the copyright

timothysc commented on test/e2e/framework/metrics/e2e_metrics.go:
```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2015 The Kubernetes Authors.
```
same

timothysc reviewed (changes requested):
/approve 
^ of this change, but please address comments and others can lgtm 

I know you didn't create this code, but man this is such a cobbled mess.  A bunch of the custom metrics stuff should really be it's own library and much cleaner imo. @kubernetes/sig-instrumentation-misc should own this cleanup!

alejandrox1 commented on test/e2e/framework/metrics/latencies.go:
```diff
@@ -0,0 +1,363 @@
+	if err != nil {
+		e2elog.Logf("Unexpected error occurred: %v", err)
+	}
+	gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
```
@andrewsykim here I replaced a framework.ExpectNoError(err) with the code for it 
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L1360-L1372

alejandrox1 replied:
cc @timstclair

alejandrox1 commented on test/e2e/framework/metrics/latencies.go:
```diff
@@ -0,0 +1,363 @@
+		if err != nil {
+			e2elog.Logf("Unexpected error occurred: %v", err)
+		}
+		gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
```
@andrewsykim replaced a framework.ExpectNoError(err) with the code for it here as well
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L1360-L1372

alejandrox1 replied:
cc @timstclair

andrewsykim replied:
Wrong Tim St Clair! cc @timothysc

alejandrox1 commented:
Updated licences for newly created files (as per git) and added a golint exception for `test/e2e/framework/metrics` because of `metrics.MetricsForE2E`.

alejandrox1 commented:
got a random error. let's try an replicate https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/79753/pull-kubernetes-e2e-gce-100-performance/1148375326611476480/

/test pull-kubernetes-e2e-gce-100-performance

oomichi commented on hack/.golint_failures:
```diff
@@ -558,6 +558,7 @@ staging/src/k8s.io/sample-apiserver/pkg/apis/wardle/v1alpha1
 staging/src/k8s.io/sample-apiserver/pkg/registry/wardle/fischer
 staging/src/k8s.io/sample-apiserver/pkg/registry/wardle/flunder
 test/e2e/common
+test/e2e/framework/metrics
```
Please fix golint failures of the file instead of adding the exception.
We've taken much time for fixing them under `test/e2e/framework` already.

alejandrox1 replied:
As i mentioned in a previous comment, the golint failures we are ignoring here are from `MetricsForE2E`. We are open to any suggestion you may have to rename that struct but otherwise the plan is to ignore them for this package.

oomichi commented:
/cc @oomichi

timothysc reviewed (approved):
/lgtm
